pkg/asset/agent/image/agentpxeimage.go

```diff
@@ -0,0 +1,173 @@
+package image
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	// pxeAssetsPath is the path where pxe image assets are created.
+	pxeAssetsPath = "pxe"
+)
+
+// AgentPXEFiles is an asset that generates the bootable image used to install clusters.
+type AgentPXEFiles struct {
+	imageReader isoeditor.ImageReader
+}
+
+var _ asset.WritableAsset = (*AgentPXEFiles)(nil)
+
+// Dependencies returns the assets on which the AgentPXEFiles asset depends.
+func (a *AgentPXEFiles) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&Ignition{},
+		&BaseIso{},
+	}
+}
+
+// Generate generates the image files for PXE asset.
+func (a *AgentPXEFiles) Generate(dependencies asset.Parents) error {
+	ignition := &Ignition{}
+	dependencies.Get(ignition)
+
+	baseImage := &BaseIso{}
+	dependencies.Get(baseImage)
+
+	ignitionByte, err := json.Marshal(ignition.Config)
+	if err != nil {
+		return err
+	}
+
+	ignitionContent := &isoeditor.IgnitionContent{Config: ignitionByte}
+
+	isoGetter := newGetIso(GetIsoPluggable)
+	isoPath, err := isoGetter.getter()
+
+	if err != nil {
+		return err
+	}
+
+	workDir, err := GetCacheDir(imageDataType)
+	if err != nil {
+		return err
+	}
+
+	err = isoeditor.Extract(isoPath, workDir)
+	if err != nil {
+		return err
+	}
+
+	irfsPath := filepath.Join(workDir, "images/pxeboot/initrd.img")
```

Comment:
Once we are extracting the files we want individually, we should write the base initrd to a tempfile, rather than putting it somewhere that could cause confusion if it got left behind after an error.
Even better would if we could stream it out of the ISO without writing it to disk, but the `NewInitRamFSStreamReader()` API currently takes a filesystem path so that isn't yet possible.